src/asset_list.rs

```diff
@@ -21,12 +24,36 @@ impl<T> Default for AssetListBase<T> {
 pub type AssetListUnchecked = AssetListBase<String>;
 pub type AssetList = AssetListBase<Addr>;
 
+impl From<Vec<AssetUnchecked>> for AssetListUnchecked {
+    fn from(assets: Vec<AssetUnchecked>) -> Self {
+        Self(assets)
+    }
+}
+
 impl From<AssetList> for AssetListUnchecked {
     fn from(list: AssetList) -> Self {
         Self(list.to_vec().iter().cloned().map(|asset| asset.into()).collect())
     }
 }
 
+impl<A, B> From<B> for AssetList
+where
+    A: Into<Asset>,
+    B: IntoIterator<Item = A>,
+{
+    fn from(list: B) -> Self {
+        Self(list.into_iter().map(|a| a.into()).collect())
+    }
+}
```

Comment:
🔥

Reply:
This would allow us to create a checked AssetList from a list of assets containing multiple entries of the same AssetInfo. This is also the case with the current implementation of  `check()`, but not with `add()`. I think we should have consistency here and vote for not allowing duplicates